git/url.go

```diff
@@ -95,20 +95,26 @@ func ParseURL(rawURL string) (u *url.URL, err error) {
 		// Not a Windows path.
 		!strings.ContainsRune(rawURL, '\\') {
 		// Support scp-like syntax for ssh protocol.
+		// We convert SCP syntax into ssh://<uri>
+		// Examples below:
+		// git@github.com:some/path.git -> ssh://github.com/some/path.git
+		// git@github.com:2222/some/path.git -> ssh://github.com:2222/some/path.git
 		index := strings.Index(rawURL, ":")
-		if index >= 0 && len(rawURL) > index && unicode.IsDigit(rune(rawURL[index+1])) {
-			next := strings.Index(rawURL[index+1:], ":")
-			if next >= 0 {
-				index = index + 1 + next
-			} else {
-				index = -1
+		if index > 0 {
+			next := strings.Index(rawURL[index+1:], "/")
+			if next > 0 {
+				// check if port is present
+				_, err := strconv.Atoi64(rawURL[index+1 : index+1+next])
```

Comment:
An account thats all numbers cannot happen, right?
`git@github.com:8472/project1.git`

Reply:
yes it can but the URL will be `git@whatever.com/2020/1.git`

Reply:
I pushed a helper command to check more cases:

```
./bin/helper git-normalization whatever.com:2222.git
host:  whatever.com
owner:
name:  2222
normalized repository: whatever.com/2222
```

if something does not make sense, let me know!